tools/lint/clang_tidy.py

```diff
@@ -0,0 +1,196 @@
+    with open(file_path, "w", encoding="utf-8") as f:
+        f.write(content)
+    logging.info("Wrote file: %s", file_path)
+
+def append_file(file_path: str, content: str):
+    """Append content to the specified file."""
+    with open(file_path, "a", encoding="utf-8") as f:
+        f.write(content + "\n")
+    logging.info("Appended content to file: %s", file_path)
+
+def generate_markdown_report(
+    errors_count: int,
+    warnings_count: int,
+    notes_count: int,
+    clang_tidy_output: str,
+    enabled_checks_content: str,
+) -> str:
+    """Generate a Markdown report section for clang-tidy results and enabled checks."""
+    report_lines = []
+    if errors_count > 0:
+        report_lines.append(
+            f"<details><summary>:fire: Clang-Tidy found :fire: {errors_count} errors, :warning: {warnings_count} warnings and :pencil2: {notes_count} notes</summary>"
+        )
+    elif warnings_count > 0:
+        report_lines.append(
+            f"<details><summary>:warning: Clang-Tidy found :warning: {warnings_count} warnings and :pencil2: {notes_count} notes</summary>"
+        )
+    elif notes_count > 0:
+        report_lines.append(
+            f"<details><summary>:pencil2: Clang-Tidy found :pencil2: {notes_count} notes</summary>"
+        )
+    else:
+        report_lines.append(
+            "<details><summary>:heavy_check_mark: Clang-Tidy found no errors, warnings or notes</summary>"
+        )
+    report_lines.append("")
+    report_lines.append("````")
+    report_lines.append(clang_tidy_output)
+    report_lines.append("````")
+    report_lines.append("</details>")
+    report_lines.append("")
+    report_lines.append(
+        "<details><summary>:information_source: Enabled checks</summary>"
+    )
+    report_lines.append("")
+    report_lines.append("````")
+    report_lines.append(enabled_checks_content)
+    report_lines.append("````")
+    report_lines.append("</details>")
+    report_lines.append("")
+    return "\n".join(report_lines)
+
+def count_occurrences(pattern: str, text: str, verbose: bool = False) -> int:
+    """Count all occurrences of a regex pattern in text using MULTILINE mode."""
+    matches = re.findall(pattern, text, re.MULTILINE)
+    if verbose:
+        logging.info("Pattern '%s' found %d matches", pattern, len(matches))
+    return len(matches)
+
+def run_enabled_checks(clang_style: str, build_dir: str, verbose: bool) -> str:
+    """
+    Run clang-tidy with --explain-config to list enabled checks.
+    Builds the command as a list.
+    """
+    cmd = [
+        "clang-tidy",
+        "--quiet",
+        f"--format-style={clang_style}",
+        "--export-fixes=clang-tidy.yaml",
+        "-p",
+        build_dir,
+        "--explain-config",
+    ]
+    if verbose:
+        logging.info("Running enabled-checks command: %s", " ".join(cmd))
+    stdout, stderr, _ = run_command(cmd, capture_output=True, check=True, text=True)
+    return stdout + stderr
```

Comment:
I'd prefer if we did not have a function for this. While I am fan of splitting into functions, so code gets easier to follow, here code just gets harder to follow. This could outside a function just have been

```
    cmd = f"clang-tidy --quiet --format-style={clang_style} --export-fixes=clang-tidy.yaml -p {build_dir} --explain-config}"
    stdout, stderr, _ = run_command(cmd)
```